workflow-core/src/commonMain/kotlin/com/squareup/workflow1/StatefulWorkflow.kt

```diff
@@ -73,7 +75,20 @@ public abstract class StatefulWorkflow<
 
   public inner class RenderContext internal constructor(
     baseContext: BaseRenderContext<PropsT, StateT, OutputT>
-  ) : BaseRenderContext<@UnsafeVariance PropsT, StateT, @UnsafeVariance OutputT> by baseContext
+  ) : BaseRenderContext<@UnsafeVariance PropsT, StateT, @UnsafeVariance OutputT> by baseContext {
+
+    public inline fun <reified CurrentStateT : Any> safeEventHandler(
```

Comment:
Note the dirty trick here: we can't make `CurrentStateT: StateT`, because `StateT` is not required to extend `Any`, but `KClass` (provider of the `safeCast()` call) requires it. So we accept any old type for `CurrentStateT` and…why not?

Reply:
```suggestion
    public inline fun <reified CurrentStateT : StateT & Any> safeEventHandler(
```

Reply:
That works! 👯